### DISCUSSION TITLE
How to use tsvector column for full-text search in Postgres?

### DISCUSSION THREAD

**Discussion by Jedliu** [mikro-orm/mikro-orm#3188] [Q&A]
Hi, how could I use the tsvector column for full-text search in Postgres?

This is how to do with TypeORM.
```typescript
@Column('tsvector', { select: false, nullable: true })
documentWithWeights?: any;
```

Thank you!

**Comment by B4nan** [comment]
You can use `columnType` to set the database type to whatever you like, you might need a custom mapped type for processing the runtime value. Alternative to `select: false` here would be `lazy: true` (such properties will be selected only if explicitly populated).

Never used this myself, but based on the postgres docs, it seems like the type represents an array, so you might also need to plug in the `ArrayType` provided by the ORM.

```ts
@Property({ columnType: 'tsvector', nullable: true })
documentWithWeights?: any;
```

**Reply by Jedliu** [reply]
Thank you for your fast reply as always. I will do the test soon.
